File: PADC7-MMNews-master/app/src/main/java/com/padcmyanmar/padc7/mmnews/data/models/NewsModel.java
package com.padcmyanmar.padc7.mmnews.data.models;

import androidx.lifecycle.LiveData;

import com.padcmyanmar.padc7.mmnews.data.vos.CommentVO;
import com.padcmyanmar.padc7.mmnews.data.vos.FavoriteVO;
import com.padcmyanmar.padc7.mmnews.data.vos.NewsVO;
import com.padcmyanmar.padc7.mmnews.data.vos.SendToVO;

import java.util.List;

public interface NewsModel {

    void addCommentNews(NewsVO news, CommentVO comment);

    void favoriteNews(NewsVO news, FavoriteVO favorite);

    void sendNewsTo(NewsVO news, SendToVO sendTo);

    //Get News.
    LiveData<List<NewsVO>> getNews(boolean isForce);

    LiveData<List<NewsVO>> loadMoreNews();

    interface NewsDelegate {

        void onSuccess(List<NewsVO> newsList);

        void onError(String msg);
    }
}

File: PADC7-MMNews-master/app/src/main/java/com/padcmyanmar/padc7/mmnews/data/models/NewsModel.kt
package com.padcmyanmar.padc7.mmnews.data.models

import androidx.lifecycle.LiveData

import com.padcmyanmar.padc7.mmnews.data.vos.CommentVO
import com.padcmyanmar.padc7.mmnews.data.vos.FavoriteVO
import com.padcmyanmar.padc7.mmnews.data.vos.NewsVO
import com.padcmyanmar.padc7.mmnews.data.vos.SendToVO

interface NewsModel {

    fun addCommentNews(news: NewsVO, comment: CommentVO)

    fun favoriteNews(news: NewsVO, favorite: FavoriteVO)

    fun sendNewsTo(news: NewsVO, sendTo: SendToVO)

    //Get News.
    fun getNews(isForce: Boolean, failure: (String,Int) -> Unit): LiveData<List<NewsVO>>

    fun loadMoreNews(failure: (String, Int) -> Unit): LiveData<List<NewsVO>>

    fun getNewsById(newsId: String): LiveData<NewsVO>

}
